### PR TITLE
Make sure that VS solution generator skips over netcoreapp processes

### PR DESCRIPTION
**Pull request by almiliMSFT**

### CODE REVIEW

File: Public/Src/IDE/Generator/IdeGenerator.cs [hunk 1]
```diff
@@ -206,7 +206,7 @@ private void ProcessRawReferences(IReadOnlyList<MsbuildFile> msbuildFiles)
                             item.SetMetadata("Name", referenceName);
                             msbuildFile.ProjectReferences.Add(referencedMsbuildFile);
                         }
-                        else if (msbuildFile is CsprojFile)
+                        else if (msbuildFile is CsprojFile) // && !reference.ToString(m_context.PathTable).Contains("NETCore.App"))
```
Comment (almiliMSFT) [inline Public/Src/IDE/Generator/IdeGenerator.cs]:
```suggestion
                        else if (msbuildFile is CsprojFile)
```

File: Public/Src/IDE/Generator/CsprojFile.cs [hunk 2]
```diff
@@ -41,6 +41,20 @@ internal sealed class CsprojFile : MsbuildFile
 
         internal override void VisitProcess(Process process, ProcessType pipCategory)
         {
+            var qualifier = Context.QualifierTable.GetQualifier(process.Provenance.QualifierId);
+
+            // only consider processes targeting Windows
+            if (qualifier.TryGetValue(Context.StringTable, "targetRuntime", out var targetRuntime) && targetRuntime != "win-x64")
```
Comment (almiliMSFT) [inline Public/Src/IDE/Generator/CsprojFile.cs]:
```suggestion
            if (qualifier.TryGetValue(Context.StringTable, "targetRuntime", out var targetRuntime) && 
                targetRuntime != "win-x64")
```

File: Public/Src/IDE/Generator/CsprojFile.cs [hunk 3]
```diff
@@ -41,6 +41,20 @@ internal sealed class CsprojFile : MsbuildFile
...
+            var qualifier = Context.QualifierTable.GetQualifier(process.Provenance.QualifierId);
+
+            // only consider processes targeting Windows
+            if (qualifier.TryGetValue(Context.StringTable, "targetRuntime", out var targetRuntime) && targetRuntime != "win-x64")
+            {
+                return;
+            }
+            
+            // HACK: skip over processes targeting netcoreapp framework
+            if (qualifier.TryGetValue(Context.StringTable, "targetFramework", out var targetFramework) && targetFramework.Contains("netcoreapp"))
```
Comment (almiliMSFT) [inline Public/Src/IDE/Generator/CsprojFile.cs]:
```suggestion
            if (qualifier.TryGetValue(Context.StringTable, "targetFramework", out var targetFramework)
                && targetFramework.Contains("netcoreapp"))
```

**Comment by dannyvv** [comment]
> I'm wondering if we should just spend the effort and make it play nice with .NET Core, so the generated solution fully works as expected?

Agree, would love for us to schedule time to emit an Sdk-Style project and handle multipel qualifeirs

**Pull request merged**